FD-2177: libffi manual install

MatthewPickinYork opened the pull request:
# Context
Our local dev `app` images aren't working! This is because the binary package for the "ffi" gem (a dependency of the `rerun` gem we install on the local-dev postgres ruby image) looks to have recently started being built on a newer platform. This meant that one of the libraries it came bundled with (`ffi_c.so`) was linked against a newer version of GLIBC than what we had on our AL2 runtime images, and complained when it couldn't find a compatible version.

# Implementation
Essentially, this change makes it so that:
a) the `libffi-devel` package is available on the image where the `rerun` gem is installed on
b) the `rerun` gem, and all of its dependencies, are built from source instead of using the pre-built binaries and libraries

The end result is that the `ffi` gem should make use of the system's own ffi libraries, and so shouldn't encounter any compatibility issues at runtime.

MatthewPickinYork committed:
Add libffi-devel to list of installed packages

To avoid having dependencies building their own libffi
libraries, ensure it's available in the system's local
libraries.

MatthewPickinYork committed:
Install ffi gem, using system libffi

As the default behaviour is for libffi to use its own
ffi-c shared object file, specifying it manually allows
it to use the system's version instead.

This should avoid compatability issues if the pre-built
libffi bundled with the gem is incompatible with the run-
time environment.

MatthewPickinYork committed:
Ensure source gems are pulled when installing rerun gem

To further minimise compatability issues with gems, ensure
that they're built from source instead of installing the
binary packages.

njm506 reviewed (approved):
Seems good, but I think you don't need `libffi-devel` for that build route

MatthewPickinYork committed:
Remove libffi-devel from package install list

As the ffi libraries will be build by the dependent gems from
source, installing them as system libraries is not required.
